hack/shoot-comparator/pkg/shoot/matcher_test.go

```diff
@@ -107,5 +134,986 @@ var _ = Describe(":: shoot matcher :: ", func() {
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Networking: &v1beta1.Networking{
+					Type:     ptr.To[string]("r-type"),
+					Nodes:    ptr.To[string]("the-nodes"),
+					Services: ptr.To[string]("svcs"),
+					Pods:     ptr.To[string]("pods"),
+					ProviderConfig: &runtime.RawExtension{
+						Raw: []byte("not important"),
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should match spec/kubernetes",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version: "1.2.3",
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version: "1.2.3",
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/kubernetes #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version: "1.2.3",
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version: "1.2.1",
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find differences in spec/kubernetes #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version:                     "1.2.3",
+					EnableStaticTokenKubeconfig: ptr.To[bool](true),
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					Version: "1.2.3",
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/kubernetes/kubeAPIServer #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should find no differences in spec/kubernetes/kubeAPIServer #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{
+						OIDCConfig: &v1beta1.OIDCConfig{
+							CABundle: ptr.To[string]("test"),
+						},
+					},
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/kubernetes/kubeAPIServer #3",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Kubernetes: v1beta1.Kubernetes{
+					KubeAPIServer: &v1beta1.KubeAPIServerConfig{
+						OIDCConfig: &v1beta1.OIDCConfig{
+							CABundle: ptr.To[string]("test"),
+						},
+					},
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/kubernetes/kubeAPIServer #4",
```

Comment:
Not all fields that are compared are included in the unit tests:
- `groupsPrefix`
- `issuerURL`
- `requiredClaims`
- `usernameClaim`
- `usernamePrefix`